### DISCUSSION TITLE
ErrorBoundary should catch custom Error

### DISCUSSION THREAD

**Discussion by alirezabonab** [remix-run/remix#4179] [Proposals]
I have logics in the `ActionFunction` method that validates the form input. If the input is invalid, it will throw a Custom Error which extends `Error`. The `CustomError` will have more detail about what was wrong and I want to get that `CustomError` in the `ErrorBoundary` method.

I looked into the source code and it seems that it serializes and deserializes the error object and in the process of serialization and deserialization it only takes the `message` and `stack` properties.

It would be awesome to be able to catch Custom Errors through `ErrorBoundaries` cause there is no other way to get details about what happened in the ErrorBoundary right now.

**Comment by kiliman** [comment]
There is an open PR to include the error `name` prop for custom errors. 

https://github.com/remix-run/remix/pull/3719

**Comment by alirezabonab** [comment]
The issue with PR is that it is not handling the `deserializeError`, And because of the the implementation is not helpful. considering serializing and deserializing does not take any types eventually it loses types and with current implementation it is not possible to make it type friendly.

**Reply by kiliman** [reply]
That's fine. I'm just saying somebody is working on supporting custom errors. The PR may need more work.
